src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt

```diff
@@ -604,10 +602,16 @@ class TestCasePanelBuilder(
         runTestButton.isEnabled = false
         isRemoved = true
 
-        ReportUpdater.removeTestCase(report, testCase, coverageVisualisationTabBuilder, generatedTestsTabData)
+        ReportUpdater.removeTestCase(report, testCase, coverageVisualisationTabFactory, generatedTestsTabData)
         GenerateTestsTabHelper.update(generatedTestsTabData)
     }
 
+    private fun previousError(code: String) = trimmedCodeToError[trimCode(code)]
+
+    private fun saveErrorForCode(code: String, error: String?) {
+        trimmedCodeToError[trimCode(code)] = error
+    }
+
```

Comment:
There are better solutions than having the code as an index in the map. The current solution is error-prone, and we cannot guarantee that the code pieces are unique.


**Do these code pieces represent the test cases generated by the LLM?** 
In this case, we have indexes that can be used as map keys as well (we would need to seal the initial index for each test case so that after test case deletion, indexes are still the same).

Reply:
created an issue